src/account_manager/builder.rs

```diff
@@ -136,7 +136,7 @@ impl AccountManagerBuilder {
         .await?;
 
         #[cfg(feature = "storage")]
-        let read_manager_builder = storage_manager.lock().await.get_account_manager_data().await.ok();
+        let read_manager_builder = Some(storage_manager.lock().await.get_account_manager_data().await?);
```

Comment:
I think you need to match on the Result because the first time when there is no db you also get an error that the data wasn't found or something and then it shouldn't return the error

Reply:
hmm, when matching the result, what should I do with the Err(x) if not returning it as done in this line?

Reply:
One of the error variant is about the entry not being found I think, so you need to convert that to None, other errors we need to actually rezturn

Reply:
Ah, so basically something like this:

```
        let read_manager_builder = match storage_manager.lock().await.get_account_manager_data().await {
            Ok(x) => Some(x),
            Err(e) => {
                if e.to_string().eq("Key not found") {
                    None
                } else {
                    return Err(e.into());
                }
            }
        };
```

bit unfortunate that the error variant isn't an enum

Reply:
Pretty sure it's `Error::RecordNotFound` but @Thoralf-M can confirm

Reply:
something like this:

```
        #[cfg(feature = "storage")]
        let read_manager_builder = match storage_manager.lock().await.get_account_manager_data().await {
            Ok(x) => Some(x),
            Err(crate::Error::IoError(_)) => None,
            Err(e) => return Err(e),
        };
```

just not sure if the IoError is the correct one atm

Reply:
It's probably not the correct one

Reply:
Yes needs to be `Error::RecordNotFound` :heavy_check_mark:

Reply:
ah im blind

Reply:
there is the     /// Record not found  RecordNotFound(String) in Error struct

Reply:
So for RecordNotFound we need to return None, but I think there is another case:
`{"type":"client","error":"no mnemonic has been stored into the Stronghold vault"}`

searching the variant for that now...

Reply:
works now